Ubuntu 20.04, python 3, pybombs 2.3.4 version bumps

jkbecker opened the pull request:
Not sure if this repo is EOL or if it should be kept up to speed, I took the liberty of bumping this to

- Ubuntu 20.04
- Python 3
- PyBOMBS 2.3.4

and made some minimal changes to the Dockerfiles where I felt it streamlined things (some of the changes are direct results of the Ubuntu bump, as it wouldn't build on 20.04 otherwise).

~~This PR is currently a **work in progress and should not be merged as-is**, as `pybombs-commondeps` **fails to build correctly**.~~

~~As far as I can tell, this is due to certain recipes installed in the commondeps Dockerfile having issues with Ubuntu 20.04 (seems related to [this discussion](https://github.com/gnuradio/pybombs/issues/573)). To me it looks like the root cause for these issues lies in (at least) the `qwt5` PyBOMBS recipe + its dependencies / related packages, which seem to refer to packages that don't work well with Ubuntu 20.04.~~

~~As I am inexperienced with recipe development, I wanted to post this anyways, maybe somoeone can pick this up on the recipe side of things to make things work?~~

~~The `pybombs-minimal` and `pybombs-prefix` images build fine on my end.~~

**This builds correctly now**. I was able to isolate the problems to the `qwt5` and `pyqwt5` packages, and implemented a workaround for those packages until the recipes are fixed (see gnuradio/gr-recipes#197).

argilo commented on pybombs-commondeps/Dockerfile:
```diff
@@ -1,16 +1,31 @@
 # (I use separate RUN commands so I can maybe have an easier time building)
 RUN pybombs install boost doxygen
 RUN pybombs install libtool autoconf automake
-RUN pybombs install qwt5 sip lxml
```
This should probably just be updated to `qwt6`.

argilo commented on pybombs-commondeps/Dockerfile:
```diff
@@ -1,16 +1,31 @@
+
 RUN pybombs config makewidth $makewidth
+
+# installing qwt5 will fail due to lack of X11 headers.
+# Temp fix until this is solved in the pybombs recipe (see gnuradio/gr-recipes#197):
+RUN apt-get update && apt-get install -y \
+	libqwt-qt5-dev \
+	python3-pyqt5.qwt
```
Echoing my comments from https://github.com/gnuradio/gr-recipes/issues/197: Is qwt5 support in Ubuntu 20.04 important? GNU Radio 3.8 uses qwt6.

jkbecker replied:
Ok, I wasn't aware 3.8 was firmly on qwt6 - will adapt the Dockerfile and see if that builds, that would get rid of the workaround.

jkbecker commented:
@argilo I removed the apt-based workaround and just switched to qwt6, builds properly now. Feel free to merge at your leisure or let me know if you think this is missing anything...?

argilo commented on pybombs-commondeps/Dockerfile:
```diff
@@ -1,16 +1,26 @@
 # This allows setting the makewidth temporarily to a higher value
 ARG makewidth=2
 
+# Temporarily set frontend to noninteractive to avoid warnings
+ARG DEBIAN_FRONTEND=noninteractive
```
Is this needed? It looks like you're not using the argument.

argilo replied:
I tested with these lines removed and everything still worked fine.

jkbecker replied:
Its a "nice to have" as in it won't break with it removed, but if you look at the apt-get output it throws a ton of warnings here that make it look like something is going wrong... I prefer to have it in there (ARG makes it only active at build-time) but I'm not insisting on it if it doesn't break w/o.

argilo replied:
Why does setting a Docker `ARG` have any effect though?

jkbecker replied:
You're right, it looks redundant given line 19... Not very clean to have it in there twice, I'll take it out.

argilo reviewed (approved):
There's a line that could still be removed, but the updated Dockerfiles appear to work well with or without it.
